assets/js/hooks/js_view.js

```diff
@@ -386,6 +391,21 @@ const JSView = {
     }
   },
 
+  getAssetsBaseUrl() {
+    // Livebook may be running behind an authentication proxy, in
+    // which case the internal assets URL is not accessible from
+    // within the iframe (served from a different origin). To
+    // workaround this, we fallback to a CDN for the assets if
+    // available for the given package.
+    return cachedPublicEndpointCheck().then((isPublicAccessible) => {
+      if (!isPublicAccessible && this.props.assetsCdnUrl) {
+        return this.props.assetsCdnUrl;
+      } else {
+        return window.location.origin + this.props.assetsBasePath;
+      }
+    });
```

Comment:
I reverted the logic, instead of checking if the CDN is accessible, we check if Livebook `/public` is accessible and only fallback to CDN if not (if the CDN points to a private package it would fail either way). This way the check is simpler and we do it only once.